### PR TITLE
Improves Docker Installation Files

### PR DESCRIPTION
**Pull request by aryanpingle**
# Improve Docker Installation Files

Aims to address issue #748. Specifically:

- [X] Renaming `CLIENT_xyz` to `OAUTH_CLIENT_xyz`
- [X] Renaming `lib.docker` to `lib.env`
- [x] Refactor docker-compose files and remove version tags from them
- [x] Adding `level: DEBUG` to traefik and traefik-forward-auth containers
- [x] Hardcoding `CLIENT_CONFIG` into installation files

## Type of Change

✅ Documentation update
✅ Refactoring

## Impact

Will improve the docker installation guide and make code more readable.

## Checklist

✅ My code adheres to the coding and style guidelines of the project.
✅ I have added tests for all the new code and any changes made to
      existing code.
✅ I have made corresponding changes to the documentation.

### CODE REVIEW

**Comment by aryanpingle** [comment]
I had to rename the variable names in `AUTHMS.drawio` as well, which made one line overflow horizontally. In [this commit](https://github.com/INTO-CPS-Association/DTaaS/pull/988/commits/08b1fd6b6ebd10f940baba4781f6e2d19840a662) I edited it using drawio.com, and here's how it looks now:

| Before | After |
|:-:|:-:|
|image|image|

It seems to have also changed some drawio metadata variables, I'm not sure if those matter:

image

**Comment by prasadtalasila** [comment]
> I had to rename the variable names in `AUTHMS.drawio` as well, which made one line overflow horizontally. In [this commit](https://github.com/INTO-CPS-Association/DTaaS/pull/988/commits/08b1fd6b6ebd10f940baba4781f6e2d19840a662) I edited it using drawio.com, and here's how it looks now:
> 
> Before	After
> image	image
> It seems to have also changed some drawio metadata variables, I'm not sure if those matter:
> 
> image

The revised figure has CLIENT_ID at the top. It should be OAUTH_CLIENT_ID.

The diff on drawio file can be ignored

**Comment by prasadtalasila** [comment]
@aryanpingle Please search for old variables in `docs/` and replace them with the new names.

**Comment by aryanpingle** [comment]
@prasadtalasila Yep I've updated all instances of the `CLIENT_ID` and `CLIENT_SECRET` environment variables in the docs as well.

I haven't modified any of the client side code like this:

```typescript
export interface OidcConfig {
  // ...
  client_id: string;
  // ...
}
```

Should I change these? On one hand they aren't gonna be confused with traefik variables, but on the other hand using the `OAUTH_` prefix would make things consistent.

File: docker/compose.dev.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ services:
       - "--entryPoints.web.address=:80"
       - "--entrypoints.web.forwardedHeaders.insecure=true"
       - "--entrypoints.web.proxyProtocol.insecure=true"
+      - "--log.level=DEBUG"
```
Comment (aryanpingle) [inline docker/compose.dev.yml:12]: @prasadtalasila I could create a blank directory `traefik-logs` and mount it to the container, and redirect General Logs and [Access Logs](https://doc.traefik.io/traefik/observability/access-logs/#access-logs) into it as well.

Should I go ahead with this? If so, where should I create the directory?

**Comment by prasadtalasila** [comment]
> @prasadtalasila Yep I've updated all instances of the `CLIENT_ID` and `CLIENT_SECRET` environment variables in the docs as well.
> 
> I haven't modified any of the client side code like this:
> 
> ```typescript
> export interface OidcConfig {
>   // ...
>   client_id: string;
>   // ...
> }
> ```
> 
> Should I change these? On one hand they aren't gonna be confused with traefik variables, but on the other hand using the `OAUTH_` prefix would make things consistent.

This code is in `client`. Please do not change it. The scope of change proposed here must be restricted to: `docker`, `docs` and `deploy`.

File: docker/compose.dev.yml (hunk 1 above)
Comment (prasadtalasila) [reply docker/compose.dev.yml:12]: Let us skip this for now.

File: deploy/docker/compose.server.secure.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@ services:
     image: intocps/libms:latest
     restart: unless-stopped
     volumes:
-      - ${DTAAS_DIR}/deploy/config/lib.docker:/dtaas/libms/.env
+      - ${DTAAS_DIR}/deploy/config/lib.env:/dtaas/libms/.env
```
Comment (prasadtalasila) [inline deploy/docker/compose.server.secure.yml:45]: This line can be removed. Please check the same with a sample installation.

Comment (prasadtalasila) [inline deploy/docker/compose.server.secure.yml:45]: same here. It can be removed as well.

File: deploy/docker/compose.server.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ services:
     image: intocps/libms:latest
     restart: unless-stopped
     volumes:
-      - ${DTAAS_DIR}/deploy/config/lib.docker:/dtaas/libms/.env
+      - ${DTAAS_DIR}/deploy/config/lib.env:/dtaas/libms/.env
```
Comment (prasadtalasila) [inline deploy/docker/compose.server.yml:23]: Same here. It could be removed.

File: docker/libms.npm.dockerfile
Comment (prasadtalasila) [inline docker/libms.npm.dockerfile:1]: please build it once to make sure that the change works as expected. BTW, does `docker/libms.dockerfile` change?

Comment (prasadtalasila) [reply docker/libms.npm.dockerfile:1]: you are right.

File: cli/tests/compose.users.test.yml
Comment (prasadtalasila) [inline cli/tests/compose.users.test.yml:1]: Please run the poetry commands, especially tests suggested in [DEVELOPER](https://github.com/INTO-CPS-Association/DTaaS/blob/feature/distributed-demo/cli/DEVELOPER.md) to make sure that the change is a non-breaking type.

**Comment by prasadtalasila** [comment]
**TODO**: issue #954 needs to be addressed in this PR

**Comment by prasadtalasila** [comment]
@aryanpingle The following changes are required in the PR as well.

1. Line-12 of lib.npm.dockerfile (please check docker buildx after the change)
2. Remove the lib.env volume (file) mapping from server installatiin files and check
3. Documentation
    * docs/admin/host.md
    * deploy/docker/SERVER.md
4. Remove `version` from `docker/compose.dev.yml`.

**Comment by aryanpingle** [comment]
- [X] Running `docker buildx build -t intocps/libms:latest -f ./docker/libms.npm.dockerfile .` builds successfully
- [ ] Checked if lib.env mapping can be removed
- [X] Documentation has been modified
- [X] Removed version from `docker/compose.dev.yml`

**Comment by aryanpingle** [comment]
# PR Summary

* Renames lib.docker to lib.env
* Removes lib.env mapping from the libms container
* Adds `OAUTH_` prefix to authorization variables
* Enables debugging for traefik & forward-auth
* Inlines the `CLIENT_CONFIG` variable directly into compose files
* Refactors docker-compose files (by removing the version tag)
* Updated the relevant markdown files

@prasadtalasila I'm not too sure, but I think the codeclimate issues are due it not understanding the [mkdocs call-outs like !!! tip](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#admonitions).

Tested this branch on a clean install: localhost and server (http and https) all run correctly. I used the integrated gitlab instance while testing in case it matters, which it likely shouldn't.

File: deploy/docker/DOCKER-ENV.md [hunk 4]
```diff
@@ -9,17 +9,20 @@ Edit all the fields according to your specific case.
...
-  | CLIENT_SECRET | 'xx' | The Secret of your server OAuth application |
+  | OAUTH_CLIENT_ID | 'xx' | The ID of your server OAuth application |
+  | OAUTH_CLIENT_SECRET | 'xx' | The Secret of your server OAuth application |
   | OAUTH_SECRET | 'random-secret-string' | Any private random string. This is a password you choose for local installation. |
   | username1 | 'user1' | The gitlab instance username of a user of DTaaS |
   | username2 | 'user2' | The gitlab instance username of a user of DTaaS |
-  | CLIENT_CONFIG | '/Users/username/DTaaS/deploy/config/client/env.js' | Full path to env.js file for client |
 
-:clipboard: Important points to note:
+<!-- markdownlint-disable MD046 -->
```
Comment (prasadtalasila) [inline deploy/docker/DOCKER-ENV.md:18]: The `tip` is not being rendered correctly in markdown viewer. Please revert to
`:clipboard: Important points to note:`

File: deploy/docker/LOCALHOST.SECURE.md [hunk 5]
```diff
@@ -94,10 +94,18 @@ Edit all the fields according to your specific case.
   |:------------|:---------------|:---------------|
   | DTAAS_DIR | '/home/Desktop/DTaaS' | Full path to the DTaaS directory. This is an absolute path with no trailing slash. |
   | username1 | 'user1' | Your gitlab username |
-  | CLIENT_CONFIG | '/home/Desktop/DTaaS/deploy/config/client/env.local.js' | Full path to env.js file for client |
 
-:clipboard: The path examples given here are for Linux OS.
-These paths can be Windows OS compatible paths as well.
+<!-- markdownlint-disable MD046 -->
```
Comment (prasadtalasila) [inline deploy/docker/LOCALHOST.SECURE.md:98]: The `tip` is not being rendered correctly in markdown viewer. Please revert to
`:clipboard: Important points to note:`

File: deploy/docker/LOCALHOST.md [hunk 6]
```diff
@@ -58,10 +58,17 @@ Edit all the fields according to your specific case.
   |:------------|:---------------|:---------------|
   | DTAAS_DIR | '/home/Desktop/DTaaS' | Full path to the DTaaS directory. This is an absolute path with no trailing slash. |
   | username1 | 'user1' | Your gitlab username |
-  | CLIENT_CONFIG | '/home/Desktop/DTaaS/deploy/config/client/env.local.js' | Full path to env.js file for client |
 
-:clipboard: The path examples given here are for Linux OS.
-These paths can be Windows OS compatible paths as well.
+<!-- markdownlint-disable MD046 -->
```
Comment (prasadtalasila) [inline deploy/docker/LOCALHOST.md:62]: The `tip` is not being rendered correctly in markdown viewer. Please revert to
`:clipboard: Important points to note:`

File: deploy/docker/SERVER.md [hunk 7]
```diff
@@ -188,7 +184,7 @@ please change,
     image: intocps/libms:latest
     restart: unless-stopped
     volumes:
-      - ${DTAAS_DIR}/deploy/config/lib.docker:/dtaas/libms/.env
+      - ${DTAAS_DIR}/deploy/config/lib.env:/dtaas/libms/.env
```
Comment (prasadtalasila) [inline deploy/docker/SERVER.md:187]: this line is no longer needed

File: docker/compose.dev.yml [hunk 8]
```diff
@@ -95,13 +95,14 @@ services:
       - PROVIDERS_GENERIC_OAUTH_AUTH_URL=${OAUTH_URL}/oauth/authorize
       - PROVIDERS_GENERIC_OAUTH_TOKEN_URL=${OAUTH_URL}/oauth/token
       - PROVIDERS_GENERIC_OAUTH_USER_URL=${OAUTH_URL}/api/v4/user
```
Comment (prasadtalasila) [inline docker/compose.dev.yml:97]: The variable names need to be changed in `docker/.env` as well

File: docker/compose.dev.yml
Comment (prasadtalasila) [inline docker/compose.dev.yml:1]: In addition, the `libms` is using `servers/lib/config/.env.default`. The developers do not have a choice of specifying config file. A new variable named `LIB_CONFIG` needs to be added to `docker/.env` file and needs to be volume mapped to `libms` service.

**Review by prasadtalasila — COMMENTED** [review]
@aryanpingle Thanks for the updates. Please see the comments.

File: deploy/docker/DOCKER-ENV.md (hunk 4 above)
Comment (aryanpingle) [reply deploy/docker/DOCKER-ENV.md:18]: Sure thing

File: deploy/docker/LOCALHOST.md (hunk 6 above)
Comment (aryanpingle) [reply deploy/docker/LOCALHOST.md]: Done. I accidentally formatted the markdowns in `docs/admin` but I guess they'll be rendered correctly, so I'll revert those specifically.

File: docker/compose.dev.yml
Comment (aryanpingle) [reply docker/compose.dev.yml:1]: Can you explain this a bit further? If I understand correctly:

1. Add a variable to `.env.default`: `LIB_CONFIG='/dev/null'` (can't map an empty string to a docker container)
2. Add this line to `compose.lib.dev.yml`:
```
    volumes:
      ...
      - ${LIB_CONFIG}:/dtaas/libms/.env
```

File: docs/admin/host.md [hunk 9]
```diff
@@ -113,33 +113,24 @@ Edit all the fields according to your specific case.
...
-       These paths can be Windows OS compatible paths as well.
-    1. The Server DNS can also be an IP address.
-       However, for proper working it is neccessary to use the
-       same convention (IP/DNS) in the `CLIENT_CONFIG` file as well.
-<!-- markdownlint-enable MD046 -->
+1. The path examples given here are for Linux OS.
+   These paths can be Windows OS compatible paths as well.
+1. The client configuration file is located at `deploy/config/client/env.js`.
+1. The Server DNS can also be an IP address.
+   However, for proper working it is neccessary to use the same convention (IP/DNS) in the client configuration file as well.
```
[the remark this answers is not in the document]
Comment (aryanpingle) [reply docs/admin/host.md]: @prasadtalasila minor issue, but it's not possible to put a line break here without it creating a new line in the bullet point list. I could instead use HTML markup like so:

```html
<ol>
   <li>
      The path examples given here are for Linux OS.<br>
      These paths can be Windows OS compatible paths as well.
   </li>
   <li>
      The client configuration file is located at
      <code>deploy/config/client/env.local.js</code>.<br>
      Edit the URLs in this file by replacing <code>http</code> with
      <code>https</code>. Beyond this, it is not necessary to modify
      this file.
   </li>
</ol>
```

Preview:

<ol>
   <li>
      The path examples given here are for Linux OS.<br>
      These paths can be Windows OS compatible paths as well.
   </li>
   <li>
      The client configuration file is located at
      <code>deploy/config/client/env.local.js</code>.<br>
      Edit the URLs in this file by replacing <code>http</code> with
      <code>https</code>. Beyond this, it is not necessary to modify
      this file.
   </li>
</ol>

---

It looks cleaner in the markdown, and makes it more customizable - but it's up to you whether it fits the conventions

**Comment by prasadtalasila** [comment]
merged via PR #1019

**Pull request closed without merging**